File: .guides/tests/diy.js

initGame = null;
turnTaken = null;
hitWallEvent = null;
hitEnergyEvent = null;
hitMonsterEvent = null;
goalReachedEvent = null;

$.getScript(window.location.origin + '/public/js/' + window.testEnv.cmd + '.js?_=' + Date.now())
.done(function (script, status) {
  
  createEmptyMaze();
    
  if(typeof initGame == 'function') {
    score = 0;
    energy = 0;
    initGame(); 
    if(score != 25 || energy != 20) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  
  if(typeof turnTaken == 'function') {
    score = 0;
    energy = 1;
    steps = 1;
    turnTaken();
    if(score != 6 || energy != 0) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  
  var _hitWallEvent = false;

  if(typeof hitWallEvent == 'function') {
    window.playSound = function (val) {
      if(val == 'bump') _hitWallEvent = true;
    }
    hitWallEvent();
    if(!_hitWallEvent) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }       
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  

  var _hitEnergyEvent = false;

  if(typeof hitEnergyEvent == 'function') {
    energy = 0;
    window.playSound = function (val) {
      if(val == 'energy') _hitEnergyEvent = true;
    }
    hitEnergyEvent();
    if(energy != 5 || !_hitEnergyEvent) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  
  var _hitMonsterEvent  = false;

  if(typeof hitMonsterEvent == 'function') {
    energy = 3;
    window.playSound = function (val) {
      if(val == 'fight') _hitMonsterEvent = true;
    }
    hitMonsterEvent();
    if(energy != 0 || !_hitMonsterEvent) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
  }  
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  
  var _goalReachedEvent = false;

  if(typeof goalReachedEvent == 'function') {
    window.playSound = function (val) {
      if(val == 'goal-reached') _goalReachedEvent = true;
    }
    goalReachedEvent();
    if(!_goalReachedEvent) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

})
.fail(function (jqxhr, settings, exception) {
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});

File: .guides/tests/diy.py

score = 0
energy = 20
steps = 0
sound = ''

def setScore(val):
  global score
  score = val

def setEnergy(val):
  global energy
  energy = val
  
def getEnergy():
  global energy
  return energy

def getSteps():
  global steps
  return steps

def playSound(val):
  global sound
  sound = val

try:
  execfile('/home/codio/workspace/public/py/diy.py')
  
  initGame()
  if score != 25 or energy != 20:
    raise ValueError('initGame is not setting the correct score or energy')
  
  score = 0
  energy = 2
  steps = 1
  turnTaken()
  if score != 6 or energy != 1:
    raise ValueError('first change and set the energy then calculat and set the score')

  hitWallEvent ()
  if sound != 'bump':
    raise bump('hitWallEvent is not playing the correct sound')

  energy = 0
  hitEnergyEvent()
  if energy != 5:
    raise ValueError('hitEnergyEvent is not changing and setting the energy value correctly')
    
  energy = 3
  hitMonsterEvent()
  if energy != 0 or sound != 'fight':
    raise ValueError('hitMonsterEvent is eeither not setting the correct energy or playing the correct sound')
    
  goalReachedEvent() 
  if sound != 'goal-reached':
    raise ValueError('goalReachedEvent is not playing the correct sound')
    
  print 'well done'
  exit(0)
except (IOError, SyntaxError) as e:
  pass
  
print 'Not quite right, try again!'
exit(1)
